plugin/documents.py

```diff
@@ -936,6 +939,9 @@ def _on_view_updated_async(self) -> None:
             self._when_selection_remains_stable_async(
                 self._do_highlights_async, first_region, after_ms=self.highlights_debounce_time)
         self.do_signature_help_async(manual=False)
+        if self._did_paste:
+            self._did_paste = False
+            self._format_on_paste_async()
```

Comment:
Think this should be before `do_signature_help_async` so that it can still trigger after paste.